test/index.js

```diff
@@ -183,3 +183,171 @@ describe('metalsmith-serve with custom indexFile', function(){
 
 });
 
+
+// not_found file serving and redirects
+describe('metalsmith-serve', function() {
+
+  var metalsmith;
+  var servePlugin;
+
+  before(function(done) {
+    metalsmith = Metalsmith("test/fixtures/site");
+
+    servePlugin = serve({
+      verbose: false,
+      "port": port,
+      "not_found": "/404.html",
+      "redirects": {
+        "/redirect_file.txt": "/index.html",
+        '/redirect_file.txt?alt=true': '/404.html'
+      }
+    });
+
+    metalsmith
+      .use(servePlugin)
+      .build(function(err) {
+        if (err) throw err;
+        done();
+      });
+
+  });
+
+  after(function(done) {
+    servePlugin.shutdown(done);
+  });
+
+  it('should serve on local port', function(done){
```

Comment:
Yes, to ensure the code still works when working when the new functionality is activated.